core/src/impl/Kokkos_Core.cpp

```diff
@@ -752,6 +802,8 @@ void Kokkos::Impl::parse_command_line_arguments(
         warn_deprecated_command_line_argument("--kokkos-ndevices",
                                               "--kokkos-num-devices");
       }
+      warn_deprecated_command_line_argument(
+          "--kokkos-num-devices", "--kokkos-map-device-id-by=mpi_rank");
```

Comment:
in the previous lines we recommend the user to use `--kokkos-num-devices`

Reply:
https://xkcd.com/1024/

Reply:
I will give another pass at cleaning parsing of cmd line args as a follow up if that's ok with you.

Reply:
sure